src/Raven.Server/RavenServer.cs

```diff
@@ -355,12 +357,23 @@ private void ListenToNewTcpConnection(TcpListener listener)
                                 tcp.PinnedBuffer
                                 ))
                             {
-                                header = JsonDeserializationClient.TcpConnectionHeaderMessage(headerJson);
+                                header = NewClient.Client.Json.JsonDeserializationClient.TcpConnectionHeaderMessage(headerJson);
                                 if (_logger.IsInfoEnabled)
                                 {
                                     _logger.Info($"New {header.Operation} TCP connection to {header.DatabaseName} from {tcpClient.Client.RemoteEndPoint}");
                                 }
                             }
+                            if (TryAuthorize(context, Configuration, tcp.Stream, header) == false)
+                            {
+                                string msg =
```

Comment:
Why are you generating the message _outside_ the if? This way, you are doing needless allocations